src/test/java/com/dpgil/pathlinker/path_linker/internal/PathLinkerModelTest.java

```diff
@@ -0,0 +1,594 @@
+		file = new File(url.getFile());
+
+		//construct the string list based on output file
+		@SuppressWarnings("resource")
+		BufferedReader readerOutput = new BufferedReader(new FileReader(file));
+
+		ans = new ArrayList<String>(); //create a new ans list for the output
+		for (String line = readerOutput.readLine(); line != null; line = readerOutput.readLine()) {
+			String[] arr = line.split("\\s+");
+			ans.add(arr[0] + " " + arr[1] + " " + arr[2]);
+		}
+
+		modelSetUp(originalNetworkUndir, EdgeWeightSetting.UNWEIGHTED, false); //create the model for algorithm
+		resultUndir = pathListToStringList(testModel.runKSP()); //construct list of paths as string to compare with ans list
+
+		//sort the lists before comparison
+		Collections.sort(ans);
+		Collections.sort(resultUndir);
+
+		assertEquals(ans, resultUndir); //test the result
+
+		//================================= Test additive edge weight setting =====================================
+		//access output file
+		url = this.getClass().getResource("/output/graph-undir-output-additive_human-interactome.txt");
+		file = new File(url.getFile());
+
+		readerOutput = new BufferedReader(new FileReader(file)); //construct the string list based on output file
+
+		ans = new ArrayList<String>(); //create a new ans list for the output
+		for (String line = readerOutput.readLine(); line != null; line = readerOutput.readLine()) {
+			String[] arr = line.split("\\s+");
+			ans.add(arr[0] + " " + arr[1] + " " + arr[2]);
+		}
+
+		modelSetUp(originalNetworkUndir, EdgeWeightSetting.ADDITIVE, false); //create the model for algorithm
+		resultUndir = pathListToStringList(testModel.runKSP()); //construct list of paths as string to compare with ans list
+
+		//sort the lists before comparison
+		Collections.sort(ans);
+		Collections.sort(resultUndir);
+
+		assertEquals(ans, resultUndir); //test the result
+
+		//================================= Test probabilities edge weight setting =====================================
+		//access output file
+		url = this.getClass().getResource("/output/graph-undir-output-probabilities_human-interactome.txt");
+		file = new File(url.getFile());
+
+		readerOutput = new BufferedReader(new FileReader(file)); //construct the string list based on output file
+
+		ans = new ArrayList<String>(); //create a new ans list for the output
+		for (String line = readerOutput.readLine(); line != null; line = readerOutput.readLine()) {
+			String[] arr = line.split("\\s+");
+			ans.add(arr[0] + " " + arr[1] + " " + arr[2]);
+		}
+
+		modelSetUp(originalNetworkUndir, EdgeWeightSetting.PROBABILITIES, false); //create the model for algorithm
+		resultUndir = pathListToStringList(testModel.runKSP()); //construct list of paths as string to compare with ans list
+
+		//sort the lists before comparison
+		Collections.sort(ans);
+		Collections.sort(resultUndir);
+
+		assertEquals(ans, resultUndir); //test the result
+	}
+
+	/**
+	 * Test: To test mixed graphs
```

Comment:
Can you add a description as to why we have this test? The "mixed" (dir-undir) graph should give the same output as treating the undirected edges as bi-directed edges. Can you add that as another assert as well?